source/opt/aggressive_dead_code_elim_pass.cpp

```diff
@@ -454,12 +321,186 @@ bool AggressiveDCEPass::AggressiveDCE(Function* func) {
       ++bi;
     }
   }
-
   return modified;
 }
 
+void AggressiveDCEPass::ProcessWorkList(Function* func) {
+  while (!worklist_.empty()) {
+    Instruction* liveInst = worklist_.front();
+    worklist_.pop();
+    AddOperandsToWorkList(liveInst);
+    MarkBlockAsLive(liveInst);
+    MarkLoadedVariablesAsLive(func, liveInst);
+    AddDecorationsToWorkList(liveInst);
+    AddDebugInstructionsToWorkList(liveInst);
+  }
+}
+
+void AggressiveDCEPass::AddDebugInstructionsToWorkList(
+    const Instruction* inst) {
+  for (auto& line_inst : inst->dbg_line_insts()) {
+    if (line_inst.IsDebugLineInst()) {
+      AddOperandsToWorkList(&line_inst);
+    }
+  }
+
+  if (inst->GetDebugScope().GetLexicalScope() != kNoDebugScope) {
+    auto* scope =
+        get_def_use_mgr()->GetDef(inst->GetDebugScope().GetLexicalScope());
+    AddToWorklist(scope);
+  }
+  if (inst->GetDebugInlinedAt() != kNoInlinedAt) {
+    auto* inlined_at = get_def_use_mgr()->GetDef(inst->GetDebugInlinedAt());
+    AddToWorklist(inlined_at);
+  }
+}
+
+void AggressiveDCEPass::AddDecorationsToWorkList(
+    const Instruction* inst) {  // Add OpDecorateId instructions that apply
```

Comment:
This comment location looks wrong in term of the code format rule.

Reply:
Yes, the IDE I use can create new functions for me, but it does odd things with comments.  I fixed up other places it did this, but I guess I miss this one.